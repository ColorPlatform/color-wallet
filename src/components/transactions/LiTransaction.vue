<template>
  <div class="li-tx">
    <div class="li-tx__icon">
      <img rel=preload 
        :style="{ borderColor: color }"
        src="~assets/images/color-logo.png"
      />
    </div>
    <div class="li-tx__content">
      <div class="li-tx__content__left">
        <div class="li-tx__content__caption">
          <p class="li-tx__content__caption__title">
            <slot name="caption" />
          </p>
        </div>
        <div class="li-tx__content__information">
          <slot name="details" />
          <span v-if="memo"> &nbsp;- {{ memo }} </span>
        </div>
      </div>
      <div v-if="!hideMetaData" class="li-tx__content__right">
        <div>
          Network Fee:&nbsp;<b>{{ fees.amount | toAtoms }}</b>
          <span>{{ fees.denom | viewDenom }}</span>
        </div>
        <div class="li-tx__content__block">
          <router-link :to="{ name: `block`, params: { height: block } }">
            Block #{{ block }}&nbsp; </router-link
          >@&nbsp;{{ date }}
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import moment from "moment"
import { atoms as toAtoms, viewDenom } from "../../scripts/num.js"

export default {
  name: `li-transaction`,
  filters: {
    toAtoms,
    viewDenom
  },
  props: {
    color: {
      type: String,
      default: null
    },
    time: {
      type: String,
      default: null
    },
    block: {
      type: Number,
      required: true
    },
    memo: {
      type: String,
      default: null
    },
    fees: {
      type: Object,
      required: true
    },
    hideMetaData: {
      type: Boolean,
      default: false
    }
  },
  computed: {
    date({ time } = this) {
      const momentTime = moment(time)
      return momentTime.format(
        `${moment().isSame(momentTime, `day`) ? `` : `MMM Do YYYY `}HH:mm:ss`
      )
    }
  }
}
</script>

<style>
.li-tx {
  display: flex;
  align-items: center;
  background: white;
  width: 100%;
  font-weight: 300;
  position: relative;
}

.li-tx .copied {
  bottom: 0;
}

a {
  color: #0a73b1;
}

a:hover {
  color: #00d6e1;
}

.li-tx b {
  font-weight: 500;
}

.li-tx__icon {
  padding: 12px 0 12px 1rem;
}

.li-tx__icon img {
  max-height: 100%;
  max-width: 52px;
  padding: 5px;
  border: 2px solid;
  display: block;
}

.li-tx__content {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: 1rem;
}

.li-tx__content__left,
.li-tx__content__right {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.li-tx__content__right {
  text-align: right;
}

.li-tx__content__information,
.li-tx__content__information > * {
  display: flex;
  flex-direction: row;
}

.li-tx__content__information,
.li-tx__content__right {
  font-size: 14px;
  color: #2d2c2c;
}

.li-tx__content__caption {
  line-height: 18px;
  font-size: 18px;
  color: black;
}

@media screen and (max-width: 767px) {
  .li-tx__content {
    flex-direction: column;
    text-align: left;
  }

  .li-tx__content__right {
    text-align: left;
    padding-top: 0.5rem;
  }
}
</style>
