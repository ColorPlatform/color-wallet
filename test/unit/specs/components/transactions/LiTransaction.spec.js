import { mount } from "@vue/test-utils"
import LiTransaction from "transactions/LiTransaction"

describe(`LiTransaction`, () => {
  let wrapper
  const propsData = {
    color: `#FFFFFF`,
    time: new Date(Date.now()).toISOString(),
    block: 500,
    // memo: `TESTING (Sent via Color Wallet)`,
    fees: {
      amount: `3421`,
      denom: `uclr`
    }
  }
  const day = 86400000

  beforeEach(() => {
    wrapper = mount(LiTransaction, {
      propsData,
      slots: {
        caption: `<span>Some Caption</span>`,
        details: `<span>Some Details</span>`
      },
      stubs: [`router-link`]
    })
  })

  it.skip(`should show a transaction item`, () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`Should print the hour only if the same day`, () => {
    expect(
      LiTransaction.computed.date({ time: new Date(Date.now()).toISOString() })
    ).toEqual(`Jan 1st 1970, 00:00:42am UTC`)
  })

  it.skip(`should show a network fee`, () => {
    expect(wrapper.text()).toContain(`0.003`)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it.skip(`should show a network fee of 0`, () => {
    wrapper.setProps({
      // fees: {
      //   amount: "0",
      //   denom: "uclr"
      // }
    })
    // Non breaking space present before fee value
    // eslint-disable-next-line no-irregular-whitespace
    expect(wrapper.text()).toContain(`Some Caption Some Details  
        Tx Hash # 
         check Copied 
        Network Fee: 
        0.003421 CLR Block #500  
       Jan 1st 1970, 00:00:42am UTC`)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  it(`Should print the datetime if we are in a different day`, () => {
    expect(
      LiTransaction.computed.date({
        time: new Date(Date.now() - day * 2).toISOString()
      })
    ).toEqual(`Dec 30th 1969, 00:00:42am UTC`)
  })
})
