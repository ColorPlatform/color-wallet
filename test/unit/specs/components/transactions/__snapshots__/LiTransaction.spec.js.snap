// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiTransaction should show a network fee 1`] = `
<div
  class="li-tx"
>
  <div
    class="li-tx__content"
  >
    <div
      class="li-tx__content__left"
    >
      <div
        class="li-tx__icon"
      >
        <img
          rel="preload"
          src="~assets/images/color.svg"
          style="border-color: #ffffff;"
        />
      </div>
       
      <div
        class="li-tx-desc"
      >
        <div
          class="li-tx__content__caption"
        >
          <p
            class="li-tx__content__caption__title alignment-res"
          >
            <span>
              Some Caption
            </span>
          </p>
        </div>
         
        <div
          class="li-tx__content__information"
        >
          <span>
            Some Details
          </span>
           
          <!---->
        </div>
         
        <div
          class="tx-hash"
        >
          
        Tx Hash # 
        
          <div
            class="address"
            id="address"
          >
            
          </div>
           
          <div
            class="copied"
          >
            <i
              class="material-icons"
            >
              check
            </i>
             
            <span>
              Copied
            </span>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="li-tx__content__right"
    >
      <div>
        
        Network Fee: 
        
        <b>
          0.003421
        </b>
         
        <span>
          CLR
        </span>
      </div>
       
      <div
        class="li-tx__content__block"
      >
        <router-link-stub
          to="[object Object]"
        >
          Block #500 
        </router-link-stub>
      </div>
       
      <div
        class="li-tx__content__block"
      >
        
       Jan 1st 1970, 00:00:42am UTC
      
      </div>
    </div>
  </div>
</div>
`;

exports[`LiTransaction should show a network fee of 0 1`] = `
<div
  class="li-tx"
>
  <div
    class="li-tx__content"
  >
    <div
      class="li-tx__content__left"
    >
      <div
        class="li-tx__icon"
      >
        <img
          rel="preload"
          src="~assets/images/color.svg"
          style="border-color: #ffffff;"
        />
      </div>
       
      <div
        class="li-tx-desc"
      >
        <div
          class="li-tx__content__caption"
        >
          <p
            class="li-tx__content__caption__title alignment-res"
          >
            <span>
              Some Caption
            </span>
          </p>
        </div>
         
        <div
          class="li-tx__content__information"
        >
          <span>
            Some Details
          </span>
           
          <!---->
        </div>
         
        <div
          class="tx-hash"
        >
          
        Tx Hash # 
        
          <div
            class="address"
            id="address"
          >
            
          </div>
           
          <div
            class="copied"
          >
            <i
              class="material-icons"
            >
              check
            </i>
             
            <span>
              Copied
            </span>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="li-tx__content__right"
    >
      <div>
        
        Network Fee: 
        
        <b>
          0.003421
        </b>
         
        <span>
          CLR
        </span>
      </div>
       
      <div
        class="li-tx__content__block"
      >
        <router-link-stub
          to="[object Object]"
        >
          Block #500 
        </router-link-stub>
      </div>
       
      <div
        class="li-tx__content__block"
      >
        
       Jan 1st 1970, 00:00:42am UTC
      
      </div>
    </div>
  </div>
</div>
`;

exports[`LiTransaction should show a transaction item 1`] = `
<div
  class="li-tx"
>
  <div
    class="li-tx__content"
  >
    <div
      class="li-tx__content__left"
    >
      <div
        class="li-tx__icon"
      >
        <img
          rel="preload"
          src="~assets/images/color.svg"
          style="border-color: #ffffff;"
        />
      </div>
       
      <div
        class="li-tx-desc"
      >
        <div
          class="li-tx__content__caption"
        >
          <p
            class="li-tx__content__caption__title alignment-res"
          >
            <span>
              Some Caption
            </span>
          </p>
        </div>
         
        <div
          class="li-tx__content__information"
        >
          <span>
            Some Details
          </span>
           
          <!---->
        </div>
         
        <div
          class="tx-hash"
        >
          
        Tx Hash # 
        
          <div
            class="address"
            id="address"
          >
            
          </div>
           
          <div
            class="copied"
          >
            <i
              class="material-icons"
            >
              check
            </i>
             
            <span>
              Copied
            </span>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="li-tx__content__right"
    >
      <div>
        
        Network Fee: 
        
        <b>
          0.003421
        </b>
         
        <span>
          CLR
        </span>
      </div>
       
      <div
        class="li-tx__content__block"
      >
        <router-link-stub
          to="[object Object]"
        >
          Block #500 
        </router-link-stub>
      </div>
       
      <div
        class="li-tx__content__block"
      >
        
       Jan 1st 1970, 00:00:42am UTC
      
      </div>
    </div>
  </div>
</div>
`;
